adafruit_max31855.py

```diff
@@ -0,0 +1,75 @@
+# The MIT License (MIT)
+#
+# Copyright (c) 2017 Radomir Dopieralski for Adafruit Industries.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+import ustruct
+
+
+class MAX31855:
+    """
+    Driver for the MAX31855 thermocouple amplifier.
+    """
+
+    def __init__(self, spi, cs):
+        self.spi = spi
+        self.cs = cs
+        self.data = bytearray(4)
+
+    def read(self, internal=False, raw=False):
+        """
+        Read the measured temperature.
+
+        If ``internal`` is ``True``, return a tuple with the measured
+        temperature first and the internal reference temperature second.
+
+        If ``raw`` is ``True``, return the values as 14- and 12- bit integers,
+        otherwise convert them to Celsuius degrees and return as floating point
+        numbers.
+        """
```

Comment:
Although the device doesn't have registers, we can still choose to use properties to represent the device state. I think that having two properties, `temperature` and `internal_temperature`, that each read and discard the other temperature is fine. The Adafruit Arduino library does [exactly that](https://github.com/adafruit/Adafruit-MAX31855-library/blob/master/Adafruit_MAX31855.cpp). Not having them in-sync is also OK by me because the device itself uses the internal temperature to adjust the other "hot end" temperature already. You shouldn't need to do it yourself, its just available in case you also want to know that value.

Regarding raw access, I agree it could be useful for some people. However, I think those people would have no trouble doing the SPI read themselves. I'd rather have this driver expose two simple properties which are easy for beginners who don't need that accuracy.